tests/static-volume_test.go

```diff
@@ -0,0 +1,158 @@
+package tests_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	controller "kubevirt.io/containerized-data-importer/pkg/controller/common"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+)
+
+var _ = Describe("checkStaticVolume tests", func() {
+	f := framework.NewFramework("transfer-test")
+
+	Context("with available PV", func() {
+		const (
+			dvName = "testdv"
+			dvSize = "1Gi"
+		)
+
+		var (
+			pvName         string
+			pvStorageClass *string
+			pvMode         *corev1.PersistentVolumeMode
+			sourceMD5      string
+		)
+
+		importDef := func() *cdiv1.DataVolume {
+			return utils.NewDataVolumeWithHTTPImport(dvName, dvSize, fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs))
+		}
+
+		uploadDef := func() *cdiv1.DataVolume {
+			return utils.NewDataVolumeForUpload(dvName, dvSize)
+		}
+
+		cloneDef := func() *cdiv1.DataVolume {
+			return utils.NewDataVolumeForImageCloning(dvName, dvSize, "baz", "foo", pvStorageClass, pvMode)
+		}
+
+		snapshotCloneDef := func() *cdiv1.DataVolume {
+			return utils.NewDataVolumeForSnapshotCloning(dvName, dvSize, "baz", "foo", pvStorageClass, pvMode)
+		}
+
+		populatorDef := func() *cdiv1.DataVolume {
+			dataSourceRef := &corev1.TypedLocalObjectReference{
+				Kind: "PersistentVolumeClaim",
+				Name: "doesnotmatter",
+			}
+			return utils.NewDataVolumeWithExternalPopulation(dvName, dvSize, *pvStorageClass, *pvMode, nil, dataSourceRef)
+		}
+
+		BeforeEach(func() {
+			By("Creating source DV")
+			// source here shouldn't matter
+			dvDef := importDef()
+			controller.AddAnnotation(dvDef, controller.AnnDeleteAfterCompletion, "false")
+			dv, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dvDef)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for source DV to succeed")
+			f.ForceBindPvcIfDvIsWaitForFirstConsumer(dv)
+			err = utils.WaitForDataVolumePhaseWithTimeout(f, f.Namespace.Name, cdiv1.Succeeded, dv.Name, 300*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+
+			pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dv.Namespace).Get(context.TODO(), dv.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			pvName = pvc.Spec.VolumeName
+			pvStorageClass = pvc.Spec.StorageClassName
+			pvMode = pvc.Spec.VolumeMode
+
+			By("Getting source MD5")
+			md5, err := f.GetMD5(f.Namespace, pvc, utils.DefaultImagePath, 0)
+			Expect(err).ToNot(HaveOccurred())
+			err = utils.DeleteVerifierPod(f.K8sClient, f.Namespace.Name)
+			Expect(err).ToNot(HaveOccurred())
+			sourceMD5 = md5
```

Comment:
Really not big deal but I think you can get rid of this and use the md5 const https://github.com/kubevirt/containerized-data-importer/blob/3788f70d086ca8e72afafea19cc3fdd6e3005d42/tests/utils/upload.go#L42-L43